vignettes/DBI-advanced.Rmd

```diff
@@ -103,36 +105,40 @@ query <- glue::glue_sql("SELECT title, {`id`} FROM film WHERE {`id`} = {param}",
 dbGetQuery(con, query)
 ```
 
-FIXME: Params work differently, don't need to quote. Only works for literal values, not for identifiers.
+Rather than performing the parameter substitution ourselves, we can push it to the DBMS by including placeholders in the query.
+Different DBMS use different placeholder schemes. MariaDB uses a `?` for each parameter and expects an ordered list of literal values for each placeholder in the query.
 
-- Placeholder syntax varies across databases. How to discover -- read documentation?
+Other DBMS may use named parameters. 
+Currently there is no list of which placeholder scheme a particular DBMS supports. 
+For now all we can recommend is consulting the documention for the DBMS driver you are using.
 
 ```{r params}
 
-safe_params <- list( 'G' )
+params <- list( 'G' )
 safe_id <- dbQuoteIdentifier(con,"rating")
 query <- paste("SELECT * FROM film WHERE ", safe_id, " = ?" )
-res <- dbSendQuery(con, query , params = safe_params )
+res <- dbSendQuery(con, query , params = params )
 dbFetch(res)
 dbClearResult(res)
 
 ```
 
-FIXME: Multiple parameters. In order of appearance if unnamed (`?`). Problem: MariaDB knows only `?` as placeholder, other databases may have named placeholders. How to demo?
+Below is an example query using multiple placeholders with the MariaDB driver.
+The placeholders are supplied as a list of values ordered to match the position of the placeholders in the query. 
```

Comment:
Does it make sense to create an RSQLite database just for one demo of named placeholders?

Reply:
Can you please add text and a FIXME for me to add the code?